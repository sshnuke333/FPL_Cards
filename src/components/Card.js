import React from 'react';

import {
    CardContent,
    Badge,
    PlayerImg,
    ImageContent,
    PlayerName,
    StatContent,
    StatButton,
    StatName,
    StatValue,
} from './Card.styles';

import { renderToStaticMarkup } from 'react-dom/server';
import { Background } from './Background';
const svgBackground = encodeURIComponent(
    renderToStaticMarkup(
        <Background gradientStart={'#3d003f'} gradientEnd={'#3f003f'} />
    )
);

export const Card = ({ player }) => {
    return (
        <CardContent>
            <ImageContent
                style={{
                    backgroundImage: `url("data:image/svg+xml,${svgBackground}"),linear-gradient(270deg, #01f5ff, #a425ff)`,
                }}
            >
                <PlayerImg
                    src={`https://resources.premierleague.com/premierleague/photos/players/250x250/p${player.code}.png`}
                    alt={`${player.first_name} ${player.second_name}'s image`}
                    onError={({ currentTarget }) => {
                        currentTarget.onerror = null;
                        currentTarget.src =
                            'https://resources.premierleague.com/premierleague/photos/players/250x250/Photo-Missing.png';
                    }}
                />
                <Badge
                    src={`https://resources.premierleague.com/premierleague/badges/100/t${player.team_code}.png`}
                    alt="Club badge"
                />
            </ImageContent>
            <PlayerName>{player.web_name.toUpperCase()}</PlayerName>
            <StatContent>
                <StatButton>
                    <StatName>POINTS</StatName>
                    <StatValue>{player.total_points}</StatValue>
                </StatButton>
                <StatButton>
                    <StatName>PP90</StatName>
                    <StatValue>
                        {(player.total_points / (player.minutes / 90)).toFixed(
                            2
                        )}
                    </StatValue>
                </StatButton>
                <StatButton>
                    <StatName>VALUE</StatName>
                    <StatValue>{parseFloat(player.value_season)}</StatValue>
                </StatButton>
                <StatButton>
                    <StatName>ICT RANK</StatName>
                    <StatValue>{player.ict_index_rank_type}</StatValue>
                </StatButton>
            </StatContent>
        </CardContent>
    );
};
