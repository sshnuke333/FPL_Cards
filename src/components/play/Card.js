import React from 'react';
import {
    CardContainer,
    CardContent,
    Badge,
    PlayerImg,
    ImageContent,
    StatContent,
    StatButton,
} from './Card.styles';

export const Card = () => {
    return (
        <CardContainer>
            <CardContent>
                <ImageContent>
                    <PlayerImg src="" alt="Player image" />
                    <Badge src="" alt="Club badge" />
                </ImageContent>
                <StatContent>
                    <StatButton />
                </StatContent>
            </CardContent>
        </CardContainer>
    );
};
