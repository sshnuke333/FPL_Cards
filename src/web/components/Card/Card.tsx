import {
    CardContent,
    Badge,
    PlayerImg,
    ImageContent,
    PlayerName,
    StatContent,
    StatButton,
    StatName,
    StatValue,
} from './Card.styles';

interface ICard {
    player: {
        code?: number;
        element_type?: number;
        first_name: string;
        second_name: string;
        web_name: string;
        now_cost?: number;
        total_points: number;
        team_code?: number;
        value_season: string;
        minutes: number;
        ict_index_rank_type: number;
    };
    margin?: string;
    display?: string;
    animate: boolean;
    zIndex: string;
    flip: boolean;
    handleStatClick: () => void;
    disabled?: boolean;
}
const Card = ({
    player,
    margin,
    display,
    zIndex,
    animate,
    flip,
    handleStatClick,
    disabled = true,
}: ICard): JSX.Element => (
    <CardContent
        margin={margin}
        display={display}
        zIndex={zIndex}
        $animate={animate}
        $flip={flip}
    >
        <ImageContent>
            <PlayerImg
                src={`https://resources.premierleague.com/premierleague/photos/players/250x250/p${player.code}.png`}
                alt={`${player.first_name} ${player.second_name}'s image`}
                onError={(event): void => {
                    const { currentTarget } = event;
                    currentTarget.onerror = null;
                    currentTarget.src =
                        'https://resources.premierleague.com/premierleague/photos/players/250x250/Photo-Missing.png';
                }}
            />
            <Badge
                src={`https://resources.premierleague.com/premierleague/badges/100/t${player.team_code}.png`}
                alt="Club badge"
            />
        </ImageContent>
        <PlayerName $smallFont={player.web_name.length > 10}>
            {player.web_name.toUpperCase()}
        </PlayerName>
        <StatContent>
            <StatButton
                id="Points"
                disabled={disabled}
                onClick={handleStatClick}
            >
                <StatName>POINTS</StatName>
                <StatValue>{player.total_points}</StatValue>
            </StatButton>
            <StatButton id="PP90" disabled={disabled} onClick={handleStatClick}>
                <StatName>PP90</StatName>
                <StatValue>
                    {(player.total_points / (player.minutes / 90)).toFixed(2)}
                </StatValue>
            </StatButton>
            <StatButton
                id="Value"
                disabled={disabled}
                onClick={handleStatClick}
            >
                <StatName>VALUE</StatName>
                <StatValue>{parseFloat(player.value_season)}</StatValue>
            </StatButton>
            <StatButton
                id="ICT-rank"
                disabled={disabled}
                onClick={handleStatClick}
            >
                <StatName>ICT RANK</StatName>
                <StatValue>{player.ict_index_rank_type}</StatValue>
            </StatButton>
        </StatContent>
    </CardContent>
);

export default Card;
