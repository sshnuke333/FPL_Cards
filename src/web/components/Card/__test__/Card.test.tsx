import { render, screen } from '@testing-library/react';
import Card from '../Card';

describe('Card component renders', () => {
    const player = {
        first_name: 'John',
        second_name: 'Smith',
        value_season: '1',
        web_name: 'Smith',
        minutes: 1,
        ict_index_rank_type: 1,
        total_points: 1,
    };
    const renderCard = (
        <Card
            player={player}
            zIndex="1"
            flip={false}
            handleStatClick={() => {}}
            animate={false}
        />
    );
    beforeEach(() => {
        render(renderCard);
    });
    it('renders correctly', () => {
        const { asFragment } = render(renderCard);
        expect(asFragment()).toMatchSnapshot();
    });

    it('a player img', () => {
        expect(
            screen.getByRole('img', {
                name: "John Smith's image",
            })
        ).toBeInTheDocument();
    });

    it("player's club badge ", () => {
        expect(screen.getByRole('img', { name: /badge/i })).toBeInTheDocument();
    });

    it('a four stat buttons', () => {
        const buttons = screen.getAllByRole('button');
        expect(buttons).toHaveLength(4);
    });
});
