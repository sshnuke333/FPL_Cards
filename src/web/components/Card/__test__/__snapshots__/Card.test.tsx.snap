// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gswNZR fDWLOU"
    display="block"
    margin="0.5rem"
  >
    <div
      class="sc-dkrFOg CabJx"
    >
      <img
        alt="John Smith's image"
        class="sc-eDvSVe fGaYPU"
        src="https://resources.premierleague.com/premierleague/photos/players/250x250/pundefined.png"
      />
      <img
        alt="Club badge"
        class="sc-hLBbgP zUJUH"
        src="https://resources.premierleague.com/premierleague/badges/100/tundefined.png"
      />
    </div>
    <div
      class="sc-jSUZER cNwpiI"
    >
      SMITH
    </div>
    <div
      class="sc-gKPRtg FTcVd"
    >
      <button
        class="sc-iBYQkv jnmNMd"
        disabled=""
        id="Points"
      >
        <span
          class="sc-ftTHYK ertyuw"
        >
          POINTS
        </span>
        <span
          class="sc-ftTHYK sc-pyfCe ertyuw vGiIq"
        >
          1
        </span>
      </button>
      <button
        class="sc-iBYQkv jnmNMd"
        disabled=""
        id="PP90"
      >
        <span
          class="sc-ftTHYK ertyuw"
        >
          PP90
        </span>
        <span
          class="sc-ftTHYK sc-pyfCe ertyuw vGiIq"
        >
          90.00
        </span>
      </button>
      <button
        class="sc-iBYQkv jnmNMd"
        disabled=""
        id="Value"
      >
        <span
          class="sc-ftTHYK ertyuw"
        >
          VALUE
        </span>
        <span
          class="sc-ftTHYK sc-pyfCe ertyuw vGiIq"
        >
          1
        </span>
      </button>
      <button
        class="sc-iBYQkv jnmNMd"
        disabled=""
        id="ICT-rank"
      >
        <span
          class="sc-ftTHYK ertyuw"
        >
          ICT RANK
        </span>
        <span
          class="sc-ftTHYK sc-pyfCe ertyuw vGiIq"
        >
          1
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
